### DISCUSSION TITLE
fleet only showing ipv6 addresses when installing osquerry package from SecurityOnion Download Page

### DISCUSSION THREAD

**Discussion by charlespctech** [Security-Onion-Solutions/securityonion#5412] [Unsupported Versions] [answered]
Is there a way to make fleet only show ipv4 addresses instead? I have verified the individual hosts I am installing the package on do have ipv4 addresses. I also tried to set the Win 10 OS's to prefer ipv4 over ipv6 then reinstalled the osquery package but I still am only showing ipv6 addresses.

![image](https://user-images.githubusercontent.com/7142801/132237777-80824c28-6a4b-4598-a504-ec0c2b52370a.png)

**Comment by charlespctech** [comment]
I guess I could modify the query fleet is using? 

![image](https://user-images.githubusercontent.com/7142801/132247432-38602928-a47c-42a2-a1b4-e9398cadd2b7.png)

**Comment by charlespctech** [comment]
I installed osquery interactive on my host and ran some test queries. 

`osquery> SELECT address AS endpoint_ip1 FROM interface_addresses where address
    ...>           not like '%:%' and address not like '127%' and address not like '169%'
    ...>           order by interface asc limit 1;
+--------------+
| endpoint_ip1 |
+--------------+
| 192.168.0.60 |
+--------------+`

I tried to modify the settings with the one that worked in my manual query by modifying the yaml config in the Fleet web interface of https://ip_so_server/fleet/settings/osquery. 

spec:
  config:
    options:
      logger_plugin: tls
      pack_delimiter: _
      logger_tls_period: 10
      distributed_plugin: tls
      disable_distributed: false
      logger_tls_endpoint: /api/v1/osquery/log
      distributed_interval: 10
      decorations_top_level: true
      distributed_tls_max_attempts: 3
      distributed_tls_read_endpoint: /api/v1/osquery/distributed/read
      distributed_tls_write_endpoint: /api/v1/osquery/distributed/write
      enable_windows_events_publisher: true
      enable_windows_events_subscriber: true
    decorators:
      always:
        - SELECT codename FROM os_version;
        - SELECT uuid AS live_query FROM system_info;
        - >-
          SELECT address AS endpoint_ip1 FROM interface_addresses where address
          not like '%:%' and address not like '127%' and address not like '169%'
          order by interface asc limit 1;
        - SELECT hardware_serial FROM system_info;
        - SELECT hostname AS hostname FROM system_info;
  overrides: {}

I saved, regenerated the kollide installer via sudo salt-call state.apply fleet.event_gen-packages and uninstalled and reinstalled the kollide launcher on my host OS however I am still only getting an IPv6 address for my host.

**Comment by defensivedepth** [comment, marked as answer]
So there was a change in FleetDM 3.10 to prefer v4 over v6 (https://github.com/fleetdm/fleet/pull/548)

Once we get FleetDM upgraded to that version + it should work as you would expect.
